src/doc.cpp

```diff
@@ -383,13 +383,20 @@ void Doc::ExportMIDI(smf::MidiFile *midiFile)
             filters.push_back(&matchStaff);
             filters.push_back(&matchLayer);
 
+            Functor prepareMIDI(&Object::PrepareMIDI);
+            PrepareMIDIParams prepareMIDIParams;
+            prepareMIDIParams.m_transSemi = transSemi;
+            prepareMIDIParams.m_currentTempo = tempo;
+            this->Process(&prepareMIDI, &prepareMIDIParams, NULL, &filters);
+
```

Comment:
Question: this is performed for each staff/layer separately. Is this an issue for arpeggios on two staves / layers?

Reply:
Right, we should pull this out of the loop.